toolkit/tools/pkg/imagecustomizerlib/liveosisobuilder.go

```diff
@@ -864,3 +894,419 @@ func createLiveOSIsoImage(buildDir, baseConfigPath string, isoConfig *imagecusto
+//     target block size. In this function, we use an optimistic approach which
+//     calculates the required disk space by multiplying the total file size by
+//     a safety factor - i.e. safe that it will be able t hold all the contents.
+//
+// inputs:
+//
+//   - 'rootDir':
+//     root folder to calculate its size.
+//   - 'safetyFactor':
+//     a multiplier used with the total number of bytes calculated.
+//
+// outputs:
+//
+//   - returns the size in mega bytes.
+func getDiskSizeEstimateInMBs(rootDir string, safetyFactor uint64) (size uint64, err error) {
+
+	sizeInBytes, err := getSizeOnDiskInBytes(rootDir)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get folder size on disk while estimating total disk size:\n%w", err)
+	}
+
+	sizeInMBs := sizeInBytes/1024/1024 + 1
+	return sizeInMBs * safetyFactor, nil
+}
+
+// createWriteableImageFromSquashfs
+//
+//   - given a squashfs image file, it creates a writeable image with two
+//     partitions, and copies the contents of the squashfs unto that writeable
+//     image.
+//   - the squashfs image file must be extracted from a previously created
+//     LiveOS iso and is specified by the LiveOSIsoBuilder.artifacts.squashfsImagePath.
+//
+// inputs:
+//
+//   - 'buildDir':
+//     path build directory (can be shared with other tools).
+//   - 'rawImageFile':
+//     the name of the raw image to create and populate with the contents of
+//     the squashfs.
+//
+// outputs:
+//
+//   - creates the specified writeable image.
+func (b *LiveOSIsoBuilder) createWriteableImageFromSquashfs(buildDir, rawImageFile string) error {
+
+	logger.Log.Infof("Creating writeable image from squashfs (%s)", b.artifacts.squashfsImagePath)
+
+	// mount squash fs
+	squashMountDir, err := ioutil.TempDir(buildDir, "tmp-squashfs-mount-")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary mount folder for squashfs:\n%w", err)
+	}
+	defer os.RemoveAll(squashMountDir)
+
+	squashMountParams := []string{b.artifacts.squashfsImagePath, squashMountDir}
+	err = shell.ExecuteLive(false, "mount", squashMountParams...)
+	if err != nil {
+		return fmt.Errorf("failed to mount squashfs:\n%w", err)
+	}
+	defer func() {
+		unmountParams := []string{squashMountDir}
+		cleanupErr := shell.ExecuteLive(false, "umount", unmountParams...)
+		if cleanupErr != nil {
+			err = fmt.Errorf("%w:\nfailed to clean-up (%s): %w", err, squashMountDir, cleanupErr)
+		}
+	}()
+
+	// estimate the new disk size
+	safeDiskSizeMB, err := getDiskSizeEstimateInMBs(squashMountDir, 2 /*safety factor*/)
+	if err != nil {
+		return fmt.Errorf("failed to calculate the disk size of %s:\n%w", squashMountDir, err)
+	}
+
+	logger.Log.Debugf("safeDiskSizeMB = %d", safeDiskSizeMB)
+
+	// define a disk layout with a boot partition and a rootfs partition
+	var maxDiskSizeMB imagecustomizerapi.DiskSize
+	maxDiskSizeMB = imagecustomizerapi.DiskSize(safeDiskSizeMB * 1024 * 1024)
```

Comment:
Add the ESP partition to the disk's max size?

Reply:
It's already included since the squashfs has both partitions included.